### PR TITLE
`comment-on-draft-pr-indicator` - Avoid appearing on wrong places

### LINKED ISSUE #7541: `comment-on-draft-pr-indicator` appears on own comments + when editing
### Description

- Following https://github.com/refined-github/refined-github/pull/7514

The banner appears in two places where it shouldn't:

- when editing comments
- to the PR author

Screenshot 3

### Example URLs

https://github.com/refined-github/sandbox/pull/7

### PR DESCRIPTION
**Pull request by kidonng**
Fix #7541

## Test URLs

- https://github.com/refined-github/sandbox/pull/7

## Screenshot

| | Before | After |
|--------|--------|--------|
| Own PR | Screenshot 2024-07-21 at 9 24 56 PM | Screenshot 2024-07-21 at 9 24 24 PM |
| New & edit comment | Screenshot 2024-07-21 at 9 25 30 PM | Screenshot 2024-07-21 at 8 38 16 PM |

### CODE REVIEW

File: source/features/comment-on-draft-pr-indicator.tsx [hunk 1]
```diff
@@ -17,7 +22,10 @@ function addDraftBanner(newCommentField: HTMLElement): void {
 }
 
 function init(signal: AbortSignal): void {
-	observe('.CommentBox file-attachment', addDraftBanner, {signal});
+	observe(`.CommentBox file-attachment:is(
+		[input="fc-new_comment_field"],
+		[input^="fc-new_inline_comment_discussion"]
+	)`, addDraftBanner, {signal});
```
Comment (fregante) [inline source/features/comment-on-draft-pr-indicator.tsx:28]: Does this selector need anything other than just the two input attribute selector? They seem pretty specific (i.e. drop commentbox file-attachment)

Comment (kidonng) [reply source/features/comment-on-draft-pr-indicator.tsx:28]: Good question, I instinctively chose to append.

File: source/features/comment-on-draft-pr-indicator.tsx [hunk 2]
```diff
@@ -5,8 +5,13 @@ import * as pageDetect from 'github-url-detection';
 import features from '../feature-manager.js';
 import observe from '../helpers/selector-observer.js';
 import createBanner from '../github-helpers/banner.js';
+import {getDiscussionAuthor, getUsername} from '../github-helpers/index.js';
 
 function addDraftBanner(newCommentField: HTMLElement): void {
+	if (getDiscussionAuthor() === getUsername()) {
```
Comment (fregante) [inline source/features/comment-on-draft-pr-indicator.tsx:11]: I'd append this to the asLongAs list, where it must be preceded. Add comment specifying that it must come after the isDraftPR check (also in asLongAs) or else it's executed on every page

File: source/github-helpers/index.ts [hunk 3]
```diff
@@ -184,3 +184,7 @@ export function scrollIntoViewIfNeeded(element: Element): void {
 	// @ts-expect-error No Firefox support https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoViewIfNeeded
 	(element.scrollIntoViewIfNeeded ?? element.scrollIntoView).call(element);
 }
+
+export function getDiscussionAuthor(): string | undefined {
```
Comment (fregante) [inline source/github-helpers/index.ts:188]: This should be called `conversation`; "discussion" in classnames is pre-"GitHub Discussions" terminology. 

Just to double-check, this selector doesn't work on actual discussions like https://github.com/parcel-bundler/parcel/discussions/9828

**Review by fregante — APPROVED** [review]
Should be good to go after fixing the last comment (unless I broke it with the include/exclude change)

**Comment by fregante** [comment]
Thanks!

**Comment by kidonng** [comment]
I probably should also fix `netiquette` since I changed to the same selector in #7516

**Pull request merged**